client/purplecaffeine/trial.py

```diff
@@ -117,13 +123,92 @@ def add_text(self, title: str, text: str):
         """
         self.texts.append((title, text))
 
-    def add_array(self, array: Union[np.ndarray, List[Any]]):
+    def add_array(self, name: str, array: Union[np.ndarray, List[Any]]):
         """Adds array to trial data.
 
         Args:
+            name: name of the array
             array: quantum circuit results
         """
-        self.arrays.append(array)
+        self.arrays.append((name, array))
+
+    def add_tag(self, tag: str):
+        """Adds any tag to trial data.
+
+        Args:
+            tag: word of your tag
+        """
+        self.tags.append(tag)
+
+    def save_trial(self):
+        """Serialize Trial into Json format and save into Backend."""
+        circuits_encoder = []
```

Comment:
If we will have our own TrialEncoder and Decoder then all this code will go away there and Trial class will be super simple and clean :)